synapse/lib/cell.py

```diff
@@ -3694,7 +3684,8 @@ async def getMirrorUrls(self):
 
         await self.ahaclient.waitready()
 
-        mirrors = await self.ahaclient.getAhaSvcMirrors(self.ahasvcname)
+        proxy = await self.ahaclient.proxy(timeout=5)
```

Comment:
This timeout should probably be user configurable?

Reply:
Fewer knobs. If someone brings it up we can add this, but 5 seconds to connect is an ocean of time. I'd rather not try to chase down "it just hangs" bugs because a user decided to set it to 9999 when they were having networking issues :)